### ISSUE TITLE
Nullable Array XML Documentation Bug

### ISSUE THREAD

**Issue by Zintom** [dotnet/roslyn#49633]
**Visual Studio 16.8.2**: 

**Steps to Reproduce**:

1: Define the following method with a single nullable array parameter.
```
/// <summary>
/// Test Documentation
/// </summary>
/// <param name="param1">This is param 1.</param>
public static void BaseMethod(string[]? param1) { }
```
2: Define two methods that inherit the documentation, one with a nullable string array parameter and one with an array of nullable string parameter (it's hard to notice but the ? operator is on the other side of the array declaration in `inheritdoc`).
```
/// <inheritdoc cref="BaseMethod(string[]?)"/>
public static void Test1(string[]? param1) { }

/// <inheritdoc cref="BaseMethod(string?[])"/>
public static void Test2(string[]? param1) { }
```

**Expected Behavior**:
`Test1` Should display the docs of `BaseMethod`
`Test2` *inheritdoc* has the NRT attribute on the wrong side of the array so should not match the `BaseMethod` and should throw compiler error CS1584 (syntactically incorrect cref attribute)

**Actual Behavior**:
`Test1` fails to show the `BaseMethod` documentation and throws compiler error CS1584
`Test2` shows the docs of `BaseMethod` even though the `inheritdoc` signature does *not* match.

**Comment by huoyaoyuan** [comment]
Their IL signature matches.
I think the right behavior is to disable nullable annotation in xml doc reference.

**Comment by jaredpar** [comment]
This behavior is "By Design". 

The XML documentation signature matching is generally about CLR signatures vs. exact C# ones. You can create similar examples by swapping out `dynamic` and `object` in the signature syntax here which is approximately what is happening for nullable reference types here.

**Comment by Zintom** [comment]
@jaredpar Sorry I didn't mention before, the intellisense autocomplete chooses the wrong signature. I think this will lead many people like myself on a wild goose chase looking for why the auto generated cref isn't working.

If it is by design then surely intellisense shouldn't autocomplete the nullable attributes?
